### DISCUSSION TITLE
Is it ok to assume that geometry.NumPoints == geometry.Coordinates.Length, for any geometry?

### DISCUSSION THREAD

**Discussion by andreasbuykx** [NetTopologySuite/NetTopologySuite#727] [Q&A] [answered]
In our code base I see a lot of geometry.Coordinates.Length but this is a very inefficient way to find out how many vertices there are as geometry.Coordinates will usually allocate an array. I think using geometry.NumPoints should be much more efficient but I want to be sure that it returns the same number. I'm a bit confused that it isn't called NumCoordinates, but that's probably because I'm new to NetTopologySuite.

**Comment by FObermaier** [comment, marked as answer]
Yes. `NumPoints` returns the number of vertices that build the geometry without building an array of `Coordinate`s.
